### PR TITLE
EOS-21119 Hare mini-provisioner reset phase does not shutdown the cluster

### PR DESCRIPTION
**Pull request by SwapnilGaonkar7**
In 'reset' phase of hare we are cleaning up the data generated. We are deleting consul KV entries generated by Hare.
But before that we need to stop Hare services as they might keep generating more data.

We are calling 'hctl shutdown' to stop services if cluster is running

Signed-off-by: Swapnil Gaonkar <swapnil.gaonkar@seagate.com>

### CODE REVIEW

**Review by knekrasov — APPROVED** [review]

**Comment by mssawant** [comment]
@SwapnilGaonkar7, Summary needs to state the problem, kindly update the summary as,
```
EOS-18907: Hare mini-provisioner reset phase does not shutdown the cluster
```

File: provisioning/miniprov/hare_mp/main.py [hunk 1]
```diff
@@ -213,6 +213,10 @@ def reset(args):
         rc = 0
         util: ConsulUtil = ConsulUtil()
 
+        if is_cluster_running():
+            logging.info('Cluster is running, calling shutdown')
```
Comment (mssawant) [inline provisioning/miniprov/hare_mp/main.py:217]: s/Cluster is running, calling shutdown/Cluster is running, shutting down

**Review by mssawant — APPROVED** [review]
Minor comments.

File: provisioning/miniprov/hare_mp/main.py (hunk 1 above)
Comment (SwapnilGaonkar7) [reply provisioning/miniprov/hare_mp/main.py]: Done

File: provisioning/miniprov/hare_mp/main.py [hunk 2]
```diff
@@ -213,6 +213,10 @@ def reset(args):
         rc = 0
         util: ConsulUtil = ConsulUtil()
 
+        if is_cluster_running():
+            logging.info('Cluster is running, shutting down')
+            shutdown_cluster()
```
Comment (suvratjoshi) [inline provisioning/miniprov/hare_mp/main.py:218]: def shutdown_cluster(skip_consul_stop: bool):
Do we need to check the default value for `skip_consul_stop`? and set to False?

**Comment by suvratjoshi** [comment]
Procedural : 
JIRA https://jts.seagate.com/browse/EOS-18907 is closed already but how this PR is being raised under that JIRA?
Sorry, as I might have missed it, is this a new requirement that we understood recently, so creating second PR?
It is not in current sprint I think. Do we need separate jira?
cc: @vaibhavparatwar

**Comment by SwapnilGaonkar7** [comment]
@suvratjoshi There is separate jira EOS-21119 for this.

File: provisioning/miniprov/hare_mp/main.py (hunk 2 above)
Comment (SwapnilGaonkar7) [reply provisioning/miniprov/hare_mp/main.py:218]: We always skip consul stop in mini provisioning

**Pull request merged**